roc.go

```diff
@@ -0,0 +1,154 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import "sort"
+
+// SortWeightedBool rearranges the data in y along with their
+// corresponding boolean weights c so that the y data are sorted.
+// The data is sorted in place.
+// c may be nil, but if c is non-nil then it must have the same
+// length as y.
+func SortWeightedBool(y []float64, c []bool) {
+	if c == nil {
+		sort.Float64s(y)
+		return
+	}
+	if len(y) != len(c) {
+		panic("stat: slice length mismatch")
+	}
+	sort.Sort(byY{
+		y: y,
+		c: c,
+	})
+}
+
+type byY struct {
```

Comment:
this can be `boolSorter` instead of `byY`, and can handle the weights == nil case. Then lastly there should be a `weightBoolSorter` that does all three.

Thanks!